dlc-manager/src/sub_channel_manager.rs

```diff
@@ -2122,6 +2132,7 @@ where
         let msg = self.ln_channel_manager.with_useable_channel_lock(
             &sub_channel_finalize.channel_id,
             counter_party,
+            None,
```

Comment:
How do we decide if we have to set this to `None`?

Reply:
I guess we set it to `Some` every time the received message includes a commitment signature which needs to be processed.

I think I should carefully review this file given this. It would be nicer if the check was included when processing a commitment signature to avoid programmer errors. But maybe the implementation would be a lot more complicated?

Reply:
I have reviewed this file carefully and it looks good!

Reply:
I originally implemented it to check every time we pass a commitment signature, but the problem is that unfortunately the call to `get_updated_funding_outpoint_commitment_signed` is not properly reverted on error, causing the duplicate tx id error in the channel monitor. I think the reason is because `get_updated_funding_outpoint_commitment_signed` calls `monitor_updating_restored`. It might be possible to make things better by changing the API to return monitor updates and only committing the updates at the end (maybe even returning them from the `with_lock` method), but that'd be quite a big change. It definitely would be cleaner though.